### PR TITLE
fix fragment bug

### PR DESCRIPTION
**Pull request by WH-0501**
This PR fix the 的fragment bug while using the rosbridge_library/test/experimental/fragmentation+srv+tcp test.

First, self.received_fragments[msg_id]["fragment_list"] in defragmentation.py has type dict, but dict doesn't support slice.

Second, response_object in test_non-ros_service_server_fragmented.py doesn't has 'service' field, and the ‘data’ field should be 'values'

### CODE REVIEW

File: rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py [hunk 1]
```diff
@@ -55,8 +55,9 @@ def calculate_service_response(request):
     service_response_data = { "data": message}                                  # service response (as defined in srv-file)
 
     response_object = { "op": "service_response",
+                        "service": service_name,
                         "id": request_object["id"],
-                        "data": service_response_data                           # put service response in "data"-field of response object (in this case it's twice "data", because response value is also named data (in srv-file)
+                        "values": service_response_data                           # put service response in "data"-field of response object (in this case it's twice "data", because response value is also named data (in srv-file)
```
Comment (nuclearsandwich) [inline rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py]: Does this comment need to be updated to refer to the new field name `"values"`?

Comment (WH-0501) [reply rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py]: This modifed is to repair service_response message error.

Comment (WH-0501) [reply rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py]: Should I resubmit it？

File: rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py [hunk 2]
```diff
@@ -169,7 +169,12 @@ def defragment(self, message):
             self.protocol.log("debug", log_msg)
 
             # Reconstruct the message
-            reconstructed_msg = ''.join(self.received_fragments[msg_id]["fragment_list"][0:message["total"]])
+            print(type(self.received_fragments[msg_id]["fragment_list"]))
```
Comment (T045T) [inline rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py]: Please remove this

File: rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py [hunk 3]
```diff
@@ -169,7 +169,12 @@ def defragment(self, message):
...
+            #reconstructed_msg = ''.join(self.received_fragments[msg_id]["fragment_list"][0:message["total"]])
```
Comment (T045T) [inline rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py]: Remove the commented code

File: rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py [hunk 4]
```diff
@@ -169,7 +169,12 @@ def defragment(self, message):
...
+            reconstructed_msg = ''
+            for key,value in self.received_fragments[msg_id]["fragment_list"].items():
```
Comment (T045T) [inline rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py]: 1. You're not checking the size bound from the message (`msg['total']`) any more
2. Please change your code back to use `''.join()`, as it's significantly faster (see https://waymoot.org/home/python_string/ )

File: rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py (hunk 1 above)
Comment (T045T) [reply rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py]: @WH-0501 You can just add the changes we've requested to the branch you created your Pull Request from, then push to it. The Pull Request will be updated automatically, no need to make a new one :)

**Comment by WH-0501** [comment]
@T045T, Thanks.  How about this change?

File: rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py [hunk 5]
```diff
@@ -56,7 +56,8 @@ def calculate_service_response(request):
 
     response_object = { "op": "service_response",
                         "id": request_object["id"],
-                        "data": service_response_data                           # put service response in "data"-field of response object (in this case it's twice "data", because response value is also named data (in srv-file)
+						"service": service_name,
```
Comment (T045T) [inline rosbridge_library/test/experimental/fragmentation+srv+tcp/test_non-ros_service_server_fragmented.py]: I'll fix the indentation in the merge commit :)

**Pull request merged**